apps/condo/domains/billing/utils/countrySpecificQRCodeParsers/ru.qrCodeParser.js

```diff
@@ -14,26 +17,26 @@ const { get } = require('lodash')
 
 /**
  * @param {string} qrStr The QR code string got from the picture
+ * @param {function} logger ValidateQRCodeService logger
  * @return {TRUQRCodeFields}
  */
 function parseRUReceiptQRCode (qrStr) {
-    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
+    const buffer = Buffer.from(qrStr, 'base64')
+    const detectedEncoding = detectEncoding(buffer)
+    const decodedRequisitesStr = convertEncoding(buffer, detectedEncoding)
+
+    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(decodedRequisitesStr)
 
     if (!matches) {
+        logger.error({ msg:'Error qr-code parsing', data: { qrStr, decodedRequisitesStr, detectedEncoding }, step: 'decoded' })
         throw new Error('Invalid QR code')
     }
+    logger.info({ msg:'Parsed qr-code', data: { qrStr, decodedRequisitesStr, detectedEncoding }, step: 'decoded' })
 
-    const requisitesStr = get(matches, ['groups', 'requisitesStr'], '')
-
-    // https://encoding.spec.whatwg.org/#koi8-r
-    // https://encoding.spec.whatwg.org/#windows-1251
-    const encodingTag = get(matches, ['groups', 'encodingTag'])
-    const encoding = get({ 1: 'cp1251', 2: 'utf-8', 3: 'koi8-r' }, encodingTag, 'utf-8')
-
-    // Skip decoding for utf string
-    const decodedRequisitesStr = encodingTag === '2' ? requisitesStr : iconv.decode(requisitesStr, encoding)
-
-    return Object.fromEntries(decodedRequisitesStr.split('|').map((part) => part.split('=', 2)))
+    return Object.fromEntries(
+        decodedRequisitesStr.slice(8) // rm tag ST
```

Comment:
Here you already has a part of qr-code with requisites. Do not need to extract it again. Please use result of regexp (requisitesStr).